### DISCUSSION TITLE
Interop Assembly for CERTPOLICYLib.dll

### DISCUSSION THREAD

**Discussion by anarabanana** [Sleepw4lker/TameMyCerts#9] [Q&A] [answered]
Hello @Sleepw4lker, 

thanks a lot for your efforts in developing this tool!

I would like to build the policy module from source by also cross-referencing the included DLLs.
I did manage to regenerate CERTCLILIB.dll by following the steps [here ](https://www.gradenegger.eu/?p=11437). However, I am struggling to track down the original library for "CERTPOLICYLib.dll". Could you please shed some light on this?

**Comment by Sleepw4lker** [comment, marked as answer]
Hello, first of all my apologies for not responding earlier.

I'm not sure if I understood your inqiry correctly. Both DLLs are provided (in binary form) within the Github repository. I assume you want to build them from scratch for security reasons.

I had to modify the Interop assembly for the CERTCLILIB.dll so that TameMyCerts couls properly ready binary values from the CA. Therefore you can compile them with ilasm.exe as described in the article.

But this does not apply for the CERTPOLICYLIB.dll which is unmodified. Therefore it sould be sufficient to create one with tlbimp.exe from a Windows Server having the CA role installed. The file you are looking for is called certpdef.dll (in analogy to the certxds.dll used for an exit moule as described in the article).

Hope I could help, and thanks a lot for your feedback.

Kind regards

**Comment by Sleepw4lker** [comment]
Hello @anarabanana I've included this in the dev branch for the next version (https://github.com/Sleepw4lker/TameMyCerts/commit/4065ba08ab5045cc35fef1b4f76b4688b831b072). You should be able to apply it easily to the current version as well.

**Comment by anarabanana** [comment]
Hello @Sleepw4lker, 

Thank you so much for your feedback! 'certpdef.dll' is indeed the file I was looking for.
